### PR TITLE
Save file in app when dowload gets triggered, so the file eq the current state

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/11072

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-pkg/src/components/AppTemplates/AppWrapper.vue [hunk 1]
```diff
@@ -466,6 +466,22 @@ export default defineComponent({
...
+     */
+    const downloadFileActionInterceptor = async (args: FileActionOptions) => {
+      if (unref(isDirty)) {
+        await save()
+        autosavePopup()
+      }
+      originalDownloadFileActionHandler(args)
+    }
+
+    unref(downloadFileActions)[0].handler = downloadFileActionInterceptor
```
Comment (JammingBen) [inline packages/web-pkg/src/components/AppTemplates/AppWrapper.vue:483]: Overwriting the handler of the action feels a bit hacky to me 🤔 Can't you just add a method that checks for the dirty state and then calls `unref(downloadFileActions)[0].handler`?

Comment (AlexAndBear) [reply packages/web-pkg/src/components/AppTemplates/AppWrapper.vue:483]: I don't get it, we need to save on download, how to "just add a method" ?

Comment (JammingBen) [reply packages/web-pkg/src/components/AppTemplates/AppWrapper.vue:483]: I thought the component calls `unref(downloadFileActions)[0].handler` directly, but that doesn't seem to be the case. Hmm. Then I don't have a better idea...

**Review by JammingBen — APPROVED** [review]

File: packages/web-pkg/src/components/AppTemplates/AppWrapper.vue (hunk 1 above)
Comment (dschmidt) [reply packages/web-pkg/src/components/AppTemplates/AppWrapper.vue:483]: It's not suuuper nice, but I don't think it's too bad. Fine for me.

The only issue I have with this, is that you `unref` in the setup function - the current code will break when the computed in the composable is reevaluated and there's a new action object.
This needs to live inside a watcher to make it more resilient - I'm sorry 😅

**Review by dschmidt — APPROVED** [review]
LGTM :)

**Pull request merged**